### PR TITLE
Convert all format strings to f-strings

### PR DESCRIPTION
**Pull request by WaylonWalker**
## Description

Converted all format strings to f-strings.  With the dropped support of python 3.5- f-strings are now the preferred option for readability.  With this large of a change please let me know your thoughts.

## Development notes

I used flynt to convert all `format` strings to `f-strings`.  I am letting CI run tests.

## Checklist

- [X] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [X] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

## Notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 * Improved documentation.
 * Fixed issue with saving a `TensorFlowModelDataset` in the HDF5 format with versioning enabled.
 * Added missing `run_result` argument in `after_pipeline_run` Hooks spec.
+* Convert all `.format()` strings to `f-strings`
```
Comment (lorenabalan) [inline RELEASE.md:23]:
```suggestion
```
This is not a user-facing change, so we don't have to add it to our release notes. :)

**Comment by deepyaman** [comment]
If this is to happen, does it make sense to:

* Add a note in the contributing guidelines saying that f-strings should be used over format calls (don't know if it makes sense to explain why % calls are still used in tests, or just let lint handle that); and/or
* Enforce f-strings over format in lint ([relevant thread/comment](https://github.com/psf/black/issues/363#issuecomment-397959927))

**Comment by lorenabalan** [comment]
> If this is to happen, does it make sense to:
> 
> * Add a note in the contributing guidelines saying that f-strings should be used over format calls (don't know if it makes sense to explain why % calls are still used in tests, or just let lint handle that); and/or

Yeah we could make a note in the style guidelines. 🤷‍♀️  Although we've not been very diligent with detailing that section. 😅 

> * Enforce f-strings over format in lint ([relevant thread/comment](https://github.com/psf/black/issues/363#issuecomment-397959927))

[That](https://github.com/asottile/pyupgrade) looks like a pretty involved plugin, and we'd probably only use the one feature. Personally, I wouldn't make it a blocking thing using precommit,  given it's just a style preference.

**Review by lorenabalan — APPROVED** [review]
Awesome!

**Review by MerelTheisenQB — APPROVED** [review]

**Pull request merged**